ssg/build_renumber.py

```diff
@@ -127,13 +130,45 @@ def _get_checkid_string(self):
     def _translate_name_to_oval_definition_id(self, name):
         return self.translator.generate_id(self._get_checkid_string(), name)
 
+    def _get_path_for_oval_document(self, name):
+        path = os.path.join(
+            self.linked_fname.replace(self.linked_fname_basename, ""),
+            "checks/oval_documents_for_each_rule/",
+        )
+        if not os.path.exists(path):
+            os.makedirs(path)
+        return os.path.join(path, name + ".xml")
+
+    def _get_list_of_names_of_oval_checks(self):
+        out = []
+        for _, rule in rules_with_ids_generator(self.xccdftree):
+            for check in rule.findall(".//{%s}check" % (XCCDF12_NS)):
+                checkcontentref = get_content_ref_if_exists_and_not_remote(check)
+                if checkcontentref is None or check.get("system") != oval_cs:
+                    continue
+
+                out.append(checkcontentref.get("name"))
+        return out
+
+    def _save_oval_document_for_each_xccdf_rule(self):
+        for name in self._get_list_of_names_of_oval_checks():
+            oval_id = self._translate_name_to_oval_definition_id(name)
+
+            ref = self.oval_document.get_all_references_of_definition(oval_id)
```

Comment:
Nitpick: I suggest rename the `ref` variable to a name that would suggest that it's a thing usually containing more than 1 item. For example: `refs` or `used_items`.